src/js/api/core-shim.js

```diff
@@ -107,15 +107,22 @@ Object.assign(CoreShim.prototype, {
 
         model.on('change:errorEvent', logError);
 
-        return this.setup.start(api).then(allPromises => {
-            if (!allPromises) {
+        return this.setup.start(api).then(setupResult => {
+            if (!setupResult.core) {
                 throw composePlayerError(null, SETUP_ERROR_PROMISE_API_CONFLICT);
             }
-            const CoreMixin = allPromises[0];
+
+            const CoreMixin = setupResult.core;
```

Comment:
Declare this at the top so you can do `if (!CoreMixin) {` instead of getting `setupResult.core` twice.